### PR TITLE
Another documentation pass

### PR DESCRIPTION
**Pull request by dshemetov**
I figured it would be faster for me to make the changes myself than to make comments in the other PR, so here is a documentation pass.

- Merged `dev` into the branch.
- A number of style consistency fixes.
- Paid attention to specify whether arguments are vectors where appropriate.
- Specified optional arguments where appropriate.
- Added @references tags to various API docs strings.
- Reworded a number of argument specifications, corrected epirange format specifications for dates vs epiweeks.
- Fixed as_of, issues, and lags arg strings.

### CODE REVIEW

File: R/endpoints.R [hunk 1]
```diff
@@ -1,16 +1,20 @@
-#' fetch AFHSB data (point data, no min/max)
+#' Fetch AFHSB data (point data, no min/max)
 #'
-#' API docs: https://cmu-delphi.github.io/delphi-epidata/api/afhsb.html
+#' @param auth Character. Authentication token.
+#' @param locations Character vector. Locations to fetch (region/state, or 3-letter country code labels).
+#' @param epiweeks Epirange. Epiweeks to fetch in the form epirange(startweek, endweek), where
```
Comment (brookslogan) [inline R/endpoints.R:5]: suggestion: link to `epirange` on first mention.  Like
```
[`epirange`]
```
if you don't care about capitalization.

File: R/endpoints.R [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
+#'   startweek and endweek are in the form YYYYWW.
+#' @param flu_types Character. Flu types to fetch (disjoint (flu1, flu2-flu1, flu3-flu2, ili-flu3)
```
Comment (brookslogan) [inline R/endpoints.R:7]: suggestion: clarify: character vector, or string?  Same for future instances of "Character."

Comment (brookslogan) [reply R/endpoints.R:7]: suggestion: put quotes around the options, e.g., `"flu2-flu1"`

File: R/endpoints.R [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
+#'   or subset (flu2, flu3, ili) flu type labels).
+#' @return An instance of epidata_call.
 #'
-#' @param auth character authentication token
-#' @param locations character locations to fetch
-#' @param epiweeks epirange epiweeks to fetch
-#' @param flu_types character flu_types to fetch
-#' @return an instance of epidata_call
+#' @references API docs: https://cmu-delphi.github.io/delphi-epidata/api/afhsb.html
```
Comment (brookslogan) [inline R/endpoints.R:11]: quibble: References appear at the bottom in pkgdown, but these API doc references are the only place where we actually describe the nature of the data set in more than a sentence (well, maybe not for this particular endpoint, but hopefully for others).

File: R/endpoints.R [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
 #'
 #' @examples
+#' \donttest{
 #' call <- pvt_afhsb(auth = "yourkey", "fl,ca", epirange(202001, 202110), "flu1,flu2-flu1")
-#' # fetch_csv(call)
+#' fetch_tbl(call)
```
Comment (brookslogan) [inline R/endpoints.R:16]: praise: using `\donttest` for all the API call tests is a nice change.  I think by default `devtools::check()` will run in CRAN mode and test these, so we can do that if we want to check them.

praise: like the `fetch_tbl` change.  Not sure anyone's going to use `fetch_csv` unless they want to save it and do things in another language.

File: R/endpoints.R [hunk 5]
```diff
@@ -988,37 +1059,31 @@ fluview_meta <- function() {
...
-#' fetch_classic(call)
-#' @param locations a character string that indicates a location such as national, HHS region, census division, most states and territories, and etc.
-#' @param Takes in the form epirange(startdate,enddate), where startdate and enddate are of the form YYYYMMDD (can be passed as string or numeric).
-#' @return an instance of epidata_call
-#'
-#' @references API Documentation: <https://cmu-delphi.github.io/delphi-epidata/api/nowcast.html>.
-
 
-#' fetch fluview data
+#' Fetch fluview data
```
Comment (brookslogan) [inline R/endpoints.R:1063]: suggestion: "FluView", "FluView ILINet", or "CDC Fluview ILINet"

File: R/endpoints.R [hunk 6]
```diff
@@ -988,37 +1059,31 @@ fluview_meta <- function() {
...
-#' @param lag optionally specify the issue lag
-#' @param auth optional authentication
-#' @return an instance of epidata_call
-#'
-#' @references API Documentation: <https://cmu-delphi.github.io/delphi-epidata/api/fluview.html>.  For more information on ILINet, see <https://gis.cdc.gov/grasp/fluview/fluportaldashboard.html>.
+#' @param regions Character vector. The locations to fetch. Can we any string IDs in national, HHS region,
+#'   census division, most states and territories, and so on. Full list link below.
+#' @param epiweeks Epirange. The epiweeks to fetch in the form epirange(startweek,endweek), where startweek
+#'   and endweek are of the form YYYYWW (string or numeric).
+#' @param issues Epirange. Optionally, specify the issues to fetch in the form epirange(startweek,endweek), where startweek
```
Comment (brookslogan) [inline R/endpoints.R:1078]: nitpick: Note whether something is optional in the same "sentence" as the type.

File: R/endpoints.R [hunk 7]
```diff
@@ -1065,19 +1130,23 @@ fluview <-
     )
   }
 
-#' fetch Google Flu Trends data
+#' Fetch Google Flu Trends data
 #'
 #' Obtains estimates of inluenza activity based on volume of certain search queries from Google.
 #'
-#' @details Google has discontinues Flu Trends, and this is now a static endpoint. Possibile input for locations can be found in <https://github.com/cmu-delphi/delphi-epidata/blob/main/labels/regions.txt>, <https://github.com/cmu-delphi/delphi-epidata/blob/main/labels/states.txt>, and <https://github.com/cmu-delphi/delphi-epidata/blob/main/labels/cities.txt>.
+#' @details Google has discontinues Flu Trends, and this is now a static endpoint. Possibile input for locations
```
Comment (brookslogan) [inline R/endpoints.R:1137]: typo: -> "discontinued"

File: R/endpoints.R [hunk 8]
```diff
@@ -815,20 +870,26 @@ pvt_dengue_sensors <- function(auth, names, locations, epiweeks) {
   )
 }
 
-#' fetch ECDC data
+#' Fetch ECDC data
 #'
 #' Obtain information on influenza-like-illness from the European Center of Disease Control.
```
Comment (brookslogan) [inline R/endpoints.R:875]: typo: "European Centre for Disease Prevention and Control"

File: R/endpoints.R [hunk 9]
```diff
@@ -863,22 +924,29 @@ ecdc_ili <- function(regions,
   )
 }
 
-#' fetch FluSurv virological data
+#' Fetch FluSurv virological data
```
Comment (brookslogan) [inline R/endpoints.R:927]: issue: -> "FluSurv-Net hospitalization data".  "Virological" suggests results of virological testing.  (e.g., `fluview_clinical`, `fluview_public`)

File: R/endpoints.R [hunk 10]
```diff
@@ -863,22 +924,29 @@ ecdc_ili <- function(regions,
...
-#' @references API Documentation: <https://cmu-delphi.github.io/delphi-epidata/api/flusurv.html>.   See also <https://gis.cdc.gov/GRASP/Fluview/FluHospRates.html>.
+#' }
+#' @param locations Character vector. Character strings indicating location.
+#' @param epiweeks Epirange. The epiweeks to fetch the form epirange(startweek,endweek), where startweek
+#'   and endweek are of the form YYYYWW (string or numeric).
+#' @param issues Epirange. Optionally, specify the issues to fetch in the form epirange(startweek,endweek), where
+#'   startweek and endweek are of the form YYYYWW (string or numeric). Mutually exclusive with `lag`.
+#' @param lag Integer. Optionally, specify the lag for the issues to fetch. An alternative to and mutually exclusive
+#'   with `issues`.
+#' @return An instance of epidata_call
```
Comment (brookslogan) [inline R/endpoints.R:946]: praise: really liking the way param and return types are noted.

suggestion: ``[`epidata_call`]``... and make that actually point to a help topic.

File: R/endpoints.R [hunk 11]
```diff
@@ -916,21 +984,24 @@ flusurv <- function(locations,
   )
 }
 
-#' fetch FluView virological data
-#'
-#' API docs: https://cmu-delphi.github.io/delphi-epidata/api/fluview_clinical.html
+#' Fetch FluView virological data
```
Comment (brookslogan) [inline R/endpoints.R:987]: suggestion: "[...] from clinical labs"

File: R/endpoints.R [hunk 12]
```diff
@@ -626,21 +652,41 @@ covidcast_meta <- function() {
...
+#'   are available for each data source.
+#' @param time_values Epirange. The dates to fetch, specified in the form epirange(startdate,enddate), where
+#'   startdate and enddate are of the form YYYYMMDD (can be passed as string or numeric).
+#' @param geo_values Character vector. Character strings that specify which geographies to return. "*" fetches
+#'   all geographies. A list of string IDs or a character vector of IDs will fetch specific locations. More
+#'   information can be found in <https://cmu-delphi.github.io/delphi-epidata/api/covidcast_geography.html>.
+#' @param as_of Epirange. Fetch the data as of a specific date. If not specified, the most recent data will
+#'   be returned. Specified in the form epirange(startdate,enddate), where startdate and enddate are of the
+#'   form YYYYMMDD (string or numeric). Cannot be used in conjunction with `issues` or `lag`.
+#' @param issues Epirange. Fetch the data as of a specific issue date. If not specified, the most recent data
```
Comment (brookslogan) [inline R/endpoints.R:678]: issue: copy-paste issue; not "the data as of [...]"

File: R/endpoints.R [hunk 13]
```diff
@@ -720,16 +766,19 @@ covidcast <-
     )
   }
 
-#' fetch Delphi's forecast
-#'
-#' API docs: https://cmu-delphi.github.io/delphi-epidata/api/delphi.html
+#' Fetch Delphi's forecast
```
Comment (brookslogan) [inline R/endpoints.R:769]: suggestion: "Fetch Delphi's ILINet forecasts"

File: R/endpoints.R [hunk 14]
```diff
@@ -777,19 +829,22 @@ dengue_nowcast <- function(locations, epiweeks) {
   )
 }
 
-#' fetch Delphi's digital surveillance sensors
-#'
-#' API docs: https://cmu-delphi.github.io/delphi-epidata/api/dengue_sensors.html
+#' Fetch Delphi's digital surveillance sensors
```
Comment (brookslogan) [inline R/endpoints.R:832]: suggestion: "[...] for dengue in PAHO member countries"

File: R/endpoints.R [hunk 15]
```diff
@@ -1177,12 +1254,12 @@ kcdc_ili <- function(regions,
   )
 }
 
-#' fetch AFHSB meta data
+#' Fetch AFHSB meta data
```
Comment (brookslogan) [inline R/endpoints.R:1257]: nitpick: "meta data" -> "metadata" in various places

File: R/endpoints.R [hunk 16]
```diff
@@ -1205,30 +1282,36 @@ pvt_meta_norostat <- function(auth) {
...
 #'
 #' @examples
+#' \donttest{
 #' call <- nidss_dengue(locations = "taipei", epiweeks = epirange(201201, 202001))
 #' fetch_classic(call)
-#' @param locations A character string that indicates region of interest or location of interest. Possible inputs include but are not limited to (1)  nationwide, (2) central, (3) eastern, (4) kaoping, (5) northern, (6) southern, and (7) taipei.
-#' @param epiweeks Takes in the form epirange(startdate,enddate), where startdate and enddate are of the form YYYYMMDD (can be passed as string or numeric).
+#' }
+#' @param locations Character vector. Specifies the locations of interest. Possible inputs include but are not limited
+#'   to (1) nationwide, (2) central, (3) eastern, (4) kaoping, (5) northern, (6) southern, and (7) taipei.
```
Comment (brookslogan) [inline R/endpoints.R:1310]: question: do we want to duplicate this information in the pkg docs + website?

File: R/endpoints.R [hunk 17]
```diff
@@ -1299,21 +1387,24 @@ nidss_flu <-
...
-#' # fetch_classic(call)
-#' @param auth authenfication token
-#' @param location location to fetch
-#' @param epiweeks epiweeks to fetch
-#' @return an instance of epidata_call
+#' fetch_classic(call)
+#' }
+#' @param auth Character. Specifies the authentication key.
+#' @param location Character vector. Specifies the locations of interest.
+#' @param epiweeks Eppirange. Specifies the epiweeks to be fetched in the form epirange(startweek,endweek), where
```
Comment (brookslogan) [inline R/endpoints.R:1403]: typo: Eppirange

Comment (brookslogan) [reply R/endpoints.R:1403]: nitpick: -> `` `epirange(startweek, endweek)` ``

File: R/endpoints.R [hunk 18]
```diff
@@ -1368,18 +1464,23 @@ nowcast <- function(locations, epiweeks) {
   )
 }
 
-#' fetch Paho Dengue
-#'
-#' API docs: https://cmu-delphi.github.io/delphi-epidata/api/paho_dengue.html
+#' Fetch Paho Dengue
```
Comment (brookslogan) [inline R/endpoints.R:1467]: typo: -> "PAHO"

File: R/endpoints.R [hunk 19]
```diff
@@ -1415,17 +1516,20 @@ paho_dengue <- function(regions,
   )
 }
 
-#' fetch Quidel data
-#'
-#' API docs: https://cmu-delphi.github.io/delphi-epidata/api/quidel.html
+#' Fetch Quidel data
```
Comment (brookslogan) [inline R/endpoints.R:1519]: suggestion: "Fetch Quidel testing data" / "Fetch Quidel COVID-19 and influenza testing data"

**Review by brookslogan — COMMENTED** [review]
Looks good in general.  I've noted a lot of nit picks which are not blocking but which I think we'd want to address soon (e.g., maybe also in km-edit).  I'm not sure I really gave this a good lookover; there are so many endpoints and it's hard to keep paying attention.  Probably bad for maintainability.  We might want to use some templates or the new support for inline computed strings in roxygen2.

**Comment by brookslogan** [comment]
Let me know if it's better to just merge & have me PR some changes for the above.

**Comment by dshemetov** [comment]
Thanks for the close look! I'll make the changes you suggest, but yea, probably just start making the changes you want in another branch next time, I don't think we need to discuss the docs too deeply at this point.

File: .gitignore [hunk 20]
```diff
@@ -6,6 +6,8 @@
 docs
 inst/doc
 /check
-.Rdata
-.httr-oauth
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply .gitignore:10]: The venv, env, .venv. env are common names for project-specific python virtual envs or user-specific configs (.env). I like using radian, which is an R terminal built in Python, so I make a venv for it.

File: R/endpoints.R (hunk 2 above)
Comment (dshemetov) [reply R/endpoints.R:7]: Removing this, so we don't duplicate information in API docs and here.

File: R/endpoints.R (hunk 3 above)
Comment (dshemetov) [reply R/endpoints.R:11]: Should we not use the @references tag then and just move them back up under the header?

File: R/endpoints.R (hunk 6 above)
Comment (dshemetov) [reply R/endpoints.R:1078]: I was just imitating how dplyr handled optional markings 🤷 , so I think I'm gonna keep as is for now

File: R/endpoints.R (hunk 16 above)
Comment (dshemetov) [reply R/endpoints.R:1310]: Definitely not, going to remove this and make the API doc links central

**Pull request merged**